drivers/volume/portworx/portworx.go

```diff
@@ -274,12 +274,17 @@ func (p *portworx) mapNodeStatus(status api.Status) storkvolume.NodeStatus {
 	}
 }
 
-func (p *portworx) getNodeLabels(hostname string) (map[string]string, error) {
-	obj, exists, err := p.store.GetByKey(hostname)
+func (p *portworx) getNodeLabels(nodeInfo *storkvolume.NodeInfo) (map[string]string, error) {
+	obj, exists, err := p.store.GetByKey(nodeInfo.ID)
 	if err != nil {
 		return nil, err
 	} else if !exists {
-		return nil, fmt.Errorf("Node %v not found in cache", hostname)
+		obj, exists, err = p.store.GetByKey(nodeInfo.Hostname)
+		if err != nil {
+			return nil, err
+		} else if !exists {
+			return nil, fmt.Errorf("Node %v not found in cache", nodeInfo.ID)
```

Comment:
nit: also log `nodeInfo.Hostname` in the error message since both ID and hostname lookups failed.